cv_next/app/feed/components/filters/valueSelect.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+
+export const DropdownInput = ({
+  placeHolder,
+  valueIds,
+  valueId,
+  onChange,
+  getValueById,
+  text,
+}: {
+  placeHolder: string;
+  valueIds: number[];
+  valueId: number | null;
+  getValueById: (id: number) => string;
+  onChange: (newValue: number | null) => void;
+  text: string;
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const isPlaceHolder = valueId === null;
+
+  const changeIsMenuOpen = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
+
+  const menuBorderStyle = isMenuOpen ? "outline-b-0 rounded-b-none" : "";
+  const selectionStyle = isMenuOpen ? "block" : "hidden";
+  return (
+    <>
+      <div
+        className={`outline-gray-40 relative flex h-2/6 w-1/6 items-center justify-center whitespace-nowrap bg-white outline-2 ${menuBorderStyle} box-border cursor-pointer rounded-md px-10 py-4`}
+        onClick={changeIsMenuOpen}
+      >
+        <div className={`${isPlaceHolder ? "text-gray-400" : "text-black"}`}>
```

Comment:
This entire div can be separate (and along the way, be more semantically meaningful in the code context):
```
const textValue = isPlaceHolder ?
    <div className="text-black">
        {`${text}: ${placeHolder}`}
    </div> :
    <div className="text-grey">
        {`${text}: ${getValueById(valueId)}`}
    </div>;

Reply:
Note that while useful and often times efficient in both code writing time and maintainability, conditional render *inside the return value* can quickly turn into a mess that harms readability. It's a fine line, but you should almost always prefer readability.
Also in this specific case the conditional operator is used twice on the same value, so you are not really utilizing the conditionality in a good maintainable way.